pkg/config/config.go

```diff
@@ -257,3 +267,25 @@ func asString(key string, target *string, values ...string) cm.ParseFunc {
 		return nil
 	}
 }
+
+// asStringSet parses the value at key as a sets.String (split by ',') into the target, if it exists.
+func asStringSet(key string, target *sets.String, values ...string) cm.ParseFunc {
```

Comment:
it might make more sense for this function to also take in a string set of possible values (i think that might make checking against the passed in set easier)

```go
func asStringSet(key string, target *sets.String, allowed sets.String) cm.ParseFunc
```

and then compare `allowed` with whatever the user has passed in.

Reply:
changed.